### PR TITLE
213 an organization can edit update their profile

### PR DESCRIPTION
**Pull request by yuricarvalhop**
# 🔗 Issue
[213](https://github.com/rubyforgood/pet-rescue/issues/213)

# ✍️ Description

Adds a new table, `organization_profile`, which connects `organization` to `location`. On top of that, the new table contains two new fields `email` and `phone_number`. 

A new controller and two new actions were introduced, along with the respective views, so that the user can edit and update this new relation.

I didnt add any integration test as it seems like we are skipping all of them. If I should add them anyways, just comment, and Ill create a follow up PR, as this is already big enough

# 📷 Screenshots/Demos

Screenshot from 2023-10-08 15-42-08

Errors: 
Screenshot from 2023-10-08 15-43-23

### CODE REVIEW

File: app/views/organizations/home/index.html.erb
Comment (Ptrboro) [inline app/views/organizations/home/index.html.erb:6]: A quick thought: by adjusting the controller inheritance, both organization and location should always be available, eliminating the need for &

File: app/controllers/organizations/organization_profiles_controller.rb
Comment (Ptrboro) [inline app/controllers/organizations/organization_profiles_controller.rb:1]: To ensure proper organization setup, could we inherit from `Organizations::BaseController`? This will provide us with some useful safeguards related to organization handling 🛡️

File: app/views/organizations/organization_profiles/_form.html.erb
Comment (Ptrboro) [inline app/views/organizations/organization_profiles/_form.html.erb:1]: We've just merged the bootstrap_form gem, which simplifies form creation significantly by auto-handling labels and validations 🚀 Could you update this form to utilize it?
Thanks!

File: app/controllers/organizations/organization_profiles_controller.rb
Comment (Ptrboro) [inline app/controllers/organizations/organization_profiles_controller.rb:2]:
```suggestion
  layout "dashboard"
  before_action :require_organization_admin
```
This should secure actions to org admins only and polish the views! :)

File: db/migrate/20231005144432_create_organization_profiles.rb
Comment (Ptrboro) [inline db/migrate/20231005144432_create_organization_profiles.rb:8]: Let's add `null: false` and `foreign_key: true` to these columns for extra robustness 🛡️

Don’t forget to run `rails db:rollback` and `rails db:migrate` (or `rails db:reset` if needed) to refresh the schema.rb after tweaking!

**Review by Ptrboro — CHANGES_REQUESTED** [review]
Fantastic work on this @yuricarvalhop! 🏅 

Leaving some comments to polish it up before merging. Ping me if you need any help or have questions.

File: app/views/organizations/organization_profiles/_form.html.erb
Comment (Ptrboro) [reply app/views/organizations/organization_profiles/_form.html.erb:1]: Btw, we can also do it in a separate PR and merge this one faster :)

File: app/controllers/organizations/organization_profiles_controller.rb
Comment (yuricarvalhop) [reply app/controllers/organizations/organization_profiles_controller.rb:1]: Very nice catch! :fire:

**Comment by yuricarvalhop** [comment]
@Ptrboro thanks for the nice comments! Made the changes you mentioned. About the form, Ill follow this PR up with another one adding bootstrap_form, as this PR is already too large

**Review by Ptrboro — APPROVED** [review]
LGTM! 🎉 

@edwinthinks would you like to take another look, or shall we go ahead and merge?

**Comment by kasugaijin** [comment]
Just testing this works
@all-contributors
please add @yuricarvalhop  for code. 
please add @Ptrboro  for code review.

**Comment by Ptrboro** [comment]
I'm going to merge it now as it's in good shape and will help unblock other work. 🚀

@edwinthinks, if you spot anything for revision, lets address it separately.

**Pull request merged**
